### ISSUE TITLE
Language model training - overfitting?

### ISSUE THREAD

**Issue by djstrong** [zalandoresearch/flair#393]
I have trained a language model and I am using it in a downstream task. I have tested models from different saving points:

- `| end of split 100 /498 | epoch   1 | time: 1811.98s | valid loss 0.532 | valid ppl    1.702 | learning rate 20.000000000`
- `| end of split 435 /498 | epoch   2 | time: 3121.42s | valid loss 0.455 | valid ppl    1.576 | learning rate 0.062500000`

The first one (with less training) obtains better F1 score. In the downstream task I use the same corpus (but annotated). As far as I know there is no way to overfit the language model, because we are using validation dataset. Any ideas?

**Comment by stefan-it** [comment]
Annealing the learning rate is done after a defined number of splits. That means if you use many splits the learning rate will decreased faster - in contrast to a lower split size.

To avoid this, just use a higher split size :)

**Comment by djstrong** [comment]
My question is: is it possible to overfit a language model if we use validation data (as random part of training data)?

**Comment by djstrong** [comment]
That's right, it can overfit to training (validation) corpus domain. The problem here is that, the downstream task is using the same text.
Ok, I see potential issue. Actually, the corpus has 4 subdomains. Training data is not of equal sizes across subdomains, but the downstream task is.

**Comment by alanakbik** [comment]
Hm interesting. Is the difference in downstream task performance very pronounced in our experiments?

Could you perhaps set up an experiment in which you write out an LM model every X splits and create a plot of perplexity vs downstream task performance?

**Comment by alanakbik** [comment]
Closing due to inactivity, but feel free to reopen if there are further questions / comments!
